File: src/SOP8/경기융합타운/WebSOPApp/WebSOPApp/Areas/SDMS/Controllers/SensorSimulatorController.cs
using Microsoft.AspNetCore.Mvc;
using dnsCommunicateSopServer;
using System.Collections;
using SDMS.IDAL;

namespace WebSOPApp.Areas.SDMS.Controllers
{
    using Request;
    using Response;

    [Area("SDMS")]
    public class SensorSimulatorController : Controller
    {
        private IDataManager m_dataManager = null;

        public SensorSimulatorController(IDataManager dataManager)
        {
            m_dataManager = dataManager;
        }

        [HttpPost]
        public IActionResult RequestData([FromBody] RequestData data)
        {
            if (data == null)
                return BadRequest();

            if (data.RequestSendSensorAlarm != null)
                return RequestSendSensorAlarm(data.RequestSendSensorAlarm);
            else if (data.RequestClearSensorAlarm != null)
                return RequestClearSensorAlarm(data.RequestClearSensorAlarm);
            else if (data.RequestAlarmList != null)
                return RequestAlarmList();

            return null;
        }

        private IActionResult RequestAlarmList()
        {
            return Ok(ResponseAlarmList.GetAlarmList(m_dataManager));
        }

        private IActionResult RequestSendSensorAlarm(RequestSendSensorAlarm data)
        {
            SopQueryManager mgr = new SopQueryManager();
            string strUrl = Startup.ConfigManager.Site.SOPWebServerURL;

            if (strUrl.EndsWith("/") == false)
                strUrl += "/";

            if (data.SensorType == (int)dnsData.Sensor.Facility.FacilityType.FIRE_SENSOR)
                strUrl += "api/FireSensor";
            else if (data.SensorType == (int)dnsData.Sensor.Facility.FacilityType.Intrusion_S1)
                strUrl += "api/SecuritySensor";
            else if (data.SensorType == (int)dnsData.Sensor.Facility.FacilityType.PSM_SENSOR)
                strUrl += "api/PSMSensor";
            else
                strUrl += "api/EtcSensor";


            ArrayList arrData = new ArrayList();

            arrData.Add(data.SensorType);
            arrData.Add(data.SensorTagInfoID);
            arrData.Add(data.SensorZoneID);
            arrData.Add(true);

            if (data.SensorType == (int)dnsData.Sensor.Facility.FacilityType.ETC)
            {
                // 기타 센서의 경우 2단계 알람으로 설정
                arrData.Add(2);
            }

            MessageResult result = new MessageResult();

            if (mgr.SendAlarmQuery_TEST(arrData, "POST", strUrl))
                result.Success = true;
            else
            {
                result.Success = false;
                result.Message = "센서신호 전달에 실패하였습니다.";
            }

            return Ok(result);
        }

        private IActionResult RequestClearSensorAlarm(RequestClearSensorAlarm data)
        {
            foreach (int nSensorZoneID in data.SensorZoneIDs)
            {
                SopQueryManager mgr = new SopQueryManager();
                string strUrl = Startup.ConfigManager.Site.SOPWebServerURL;

                if (strUrl.EndsWith("/") == false)
                    strUrl += "/";

                if (data.SensorType == (int)dnsData.Sensor.Facility.FacilityType.FIRE_SENSOR)
                    strUrl += "api/FireSensor";
                else
                    strUrl += "api/EtcSensor";

                ArrayList arrData = new ArrayList();

                arrData.Add(data.SensorType);
                arrData.Add(data.SensorTagInfoID);
                arrData.Add(nSensorZoneID);
                arrData.Add(false);

                if (mgr.SendAlarmQuery_TEST(arrData, "POST", strUrl) == false)
                {
                    return Ok(new MessageResult(false, "센서신호 전달에 실패하였습니다."));
                }
            }

            return Ok(new MessageResult(true, ""));
        }
    }
}

File: src/SOP8/광양/WebSOPApp/WebSOPApp/ClientApp/src/SensorSimulator/services/sensorSimulatorController.js
import { SensorSimulatorJsonManager } from "./sensorSimulatorJsonManager";

export class SensorSimulatorController {
    static async sendAlarm(sensorType, sensorZoneNo) {
        try {
            const jsonData = SensorSimulatorJsonManager.makeRequestSendSensorAlarm(sensorType, sensorZoneNo);

            const response = await fetch('api/SensorSimulator/RequestSendSensorAlarm', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json;charset=utf-8'
                },
                body: jsonData
            });

            if (response.ok) {
                const result = await response.json();

                if (result.success) {
                    return [true, ""];
                }
                else {
                    return [false, result.message];
                }
            }
        }
        catch (e) {
            console.log(e);
        }

        return [false, "sendAlarm 실패"];
    }

    static async clearAlarm(sensorType, sensorTagInfoID, sensorZoneIDs) {
        try {
            const jsonData = SensorSimulatorJsonManager.makeRequestClearSensorAlarm(sensorType, sensorTagInfoID, sensorZoneIDs);

            const response = await fetch('SDMS/SensorSimulator/RequestData', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json;charset=utf-8'
                },
                body: jsonData
            });

            if (response.ok) {
                const result = await response.json();

                if (result.success) {
                    return [result.alarmDatas, ""];
                }
                else {
                    return [null, result.message];
                }
            }
        }
        catch (e) {
            console.log(e);
        }

        return [null, "sendAlarm 실패"];
    }

    static async requestAlarmList() {
        try {
            const jsonData = SensorSimulatorJsonManager.makeRequestAlarmList();

            const response = await fetch('SDMS/SensorSimulator/RequestData', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json;charset=utf-8'
                },
                body: jsonData
            });

            if (response.ok) {
                const result = await response.json();

                if (result.success) {
                    return [result.alarms, ""];
                }
                else {
                    return [null, result.message];
                }
            }
        }
        catch (e) {
            console.log(e);
        }

        return [null, "requestAlarmList 실패"];
    }
}